### ISSUE TITLE
Cannot add playlist to queue in offline mode.

### ISSUE THREAD

**Issue by LxtteDev** [jmshrv/finamp#567]
As the title suggests, the add to queue or play next option does not work for playlists when in offline mode.  I download a playlist, go into offline mode, and hold down on the list card, press "Play Next", no message appears, and no tracks on that playlist are added to the queue.  If I add the tracks on the playlist one-by-one it still work, there is just no ability to add the entire playlist.  Below is the only log left behind when trying to add to the queue.

```
[Chopper/INFO] 2023-12-20 18:05:06.203044: --> GET BASEURL/Playlists/d6fbbbde17859fd553725d9c7c790213/Items?UserId=c707889a33e746efbe613cc4d1a1af43&IncludeItemTypes=Audio&ParentId=d6fbbbde17859fd553725d9c7c790213&Recursive=true&Fields=parentId%2CindexNumber%2CsongCount%2CchildCount%2CproviderIds%2Cgenres%2Ctags%2CEtag%2CalbumPrimaryImageTag%2CparentPrimaryImageItemId
Content-Type: application/json
X-Emby-Authorization: MediaBrowser UserId="c707889a33e746efbe613cc4d1a1af43", Client="Finamp", Device="iPhone", DeviceId="63E5B941-115A-4CB2-A80A-6857112CEEF1", Version="0.6.19"
X-Emby-Token: TOKEN
content-length: 0
--> END GET

null
```

**Info**
**Finamp Version:** 0.6.19
**Device:** Iphone
**Device Version:** 16.6.1

**Comment by LxtteDev** [comment]
I looked more at the documentation I could find, and tried randomly mashing buttons on the app and I think I found something.  Chopper talks to the HTTP server so I tried connecting to the server while still in offline mode and adding the queue works.  In other words, offline mode only works when the server is running and you can connect to it.

**Comment by LxtteDev** [comment]
I believe [this is the code](https://github.com/jmshrv/finamp/blob/67679315e14cd9d2369451f767f758b27529c605/lib/components/MusicScreen/album_item.dart#L192C13-L206C21) responsible for adding the playlist to the queue.  This would explain why adding the individual tracks would work, because the same function [but for the tracks](https://github.com/jmshrv/finamp/blob/67679315e14cd9d2369451f767f758b27529c605/lib/components/AlbumScreen/song_list_tile.dart#L346C11-L354C19) does not call `jellyfinApiHelper.getItems()` .

**Comment by LxtteDev** [comment]
After doing even more digging, I think that playlist tracks can still be shown in offline mode because of [this snippet](https://github.com/jmshrv/finamp/blob/67679315e14cd9d2369451f767f758b27529c605/lib/screens/album_screen.dart#L46C11-L58C12).  I'm not 100% certain, and I have not the slightest clue on how to use this to patch this bug, but I think someone more familiar with the codebase could help out?

**Comment by Chaphasilor** [comment]
Thanks for the report! I've implemented a fix and opened a Pull Request for it at #569, so it should be part of the next update :)
